### PR TITLE
pythonPackages.ablog: init at 0.8.4

### PR DESCRIPTION
**Pull request by massimo-zaniboni**
added also the necessary package

pythonPackages.invoke: init at 0.17.0

###### Motivation for this change

I need to use the package in production, so I added and tested it.

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [X] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by FRidh** [comment]
Please read the header of the file you modified.

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -363,6 +363,28 @@ in {
...
   # packages defined here
 
+  ablog = buildPythonPackage rec {
+    pname = "ablog";
+    version = "0.8.4";
+    name = pname + "-" + version;
+
+    propagatedBuildInputs = with self; [ sphinx alabaster invoke dateutil werkzeug ];
+
+    src = pkgs.fetchurl {
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: fetchPypi

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -363,6 +363,28 @@ in {
...
+      sha256 = "1aapxswzlr920yg0k324xwdvsrdx8kydpxg8rrha1xbq62ll7fdb";
+    };
+
+    # Tests not distributed
+    doCheck = false;
+
+    meta = {
+      description = "A Sphinx extension that converts any documentation or personal website project into a full-fledged blog.";
+      license = licenses.mit;
+      homepage = http://ablog.readthedocs.org/;
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: are you going to maintain this package?

**Review by FRidh — CHANGES_REQUESTED** [review]

**Comment by massimo-zaniboni** [comment]
About travis error: I will fix the merge request within few days, applying this time all the necessary test commands, as listed in the error messages. So I hope to be 100% correct the next time. Sorry! 

BTW I followed the instructions on http://nixos.org/nixpkgs/manual/#chap-submitting-changes but they are not in par with what I understood here/now. Maybe I will try to update the manual too.

**Comment by massimo-zaniboni** [comment]
> Please read the header of the file you modified.

ABlog is an application / Python cli tool so (if I understood correctly) I need to move away from this module that contains only libraries, and build with ``buildPythonApplication`` in the proper module for stand-alone applications.

BTW as partial excuse, in this module there is also Sphinx, from which ABlog is derived, and so I considered (faulty) as the right place. Sorry.

I will update within few days, and I will integrate also the other change requests.

**Comment by massimo-zaniboni** [comment]
@FRidh I decided to abort the PR because I will not use ABlog in production, and probably this PR is not useful to others. 

Many thanks for the review, and sorry for the waste of time. In any case I understood better something about Nix.

**Pull request closed without merging**